sdk/src/types/block/core/block.rs

```diff
@@ -57,6 +64,11 @@ impl UnsignedBlock {
         [self.header.hash(), self.body.hash()].concat()
     }
 
+    /// Return the Blake2b hash of the block's signing input.
+    pub fn signing_hash(&self) -> BlockSigningHash {
+        BlockSigningHash::new(Blake2b256::digest(self.signing_input()).into())
```

Comment:
This shouldn't be hashed?

Reply:
I have thought about emitting the block event with the raw signing input, perhaps that would be better?

Reply:
Contrary to TransactionSigningHash, the protocol doesn't care about BlockSigningHash so I'd say maybe we should emit the input yes

Reply:
Ledger nano should also show the raw signing input and you should be able to compare it with the data from the event, otherwise there is no point in having it at all